Second try

ArtemKarpenko88 opened the pull request.

Commit:
Second try

pomegranate282922 commented on src/main/java/date/CreateRandomBun.java:
```diff
@@ -0,0 +1,15 @@
+package date;
```
Наверное, имелось в виду data.
Date = дата
Data = данные

pomegranate282922 commented on src/test/java/unit_test/BurgerTest.java:
```diff
@@ -0,0 +1,121 @@
+    @Mock
+    Bun bun;
+    @Mock
+    Burger burger;
```
Зачем аннотация мок на бургере? Нужно исправить

ArtemKarpenko88 replied:
Может хотели сказать 
@Mock
    CreateRandomBun createRandomBun;
    Я ее не сипользую, удалил а burger используется 
![image](https://user-images.githubusercontent.com/124075420/236195985-d177e116-df34-4ddb-ae21-103424a780aa.png)

pomegranate282922 replied:
Давай по-другому: зачем нужен мок бургера?

pomegranate282922 replied:
Здесь есть принципиальный момент. Мокировать то, что ты сейчас тестируешь - бессмысленно. Мы тестируем реальное поведение методов на реальном объекте. Сейчас ты тестируешь бургер. Мокировать бургер - неправильно.
Нужно исправить

pomegranate282922 commented on src/test/java/unit_test/BurgerTest.java:
```diff
@@ -0,0 +1,121 @@
+    public void setUpTest() {
+        burger = new Burger();
+        burger.setBuns(bun);
+        randomIngredient = Mockito.mock(Ingredient.class);
```
Зачем ещё раз мок указывать, если есть аннотация? Нужно исправить

pomegranate282922 replied:
Не исправлено. У тебя уже есть аннотация мок на ингредиенте, не нужно ещё раз делать мок. Нужно исправить

pomegranate282922 replied:
Не исправлено. Ты видишь этот коммент? Не надо дважды(!) - аннотацией и методом создавать мок.

pomegranate282922 commented on src/test/java/unit_test/BurgerTest.java:
```diff
@@ -0,0 +1,121 @@
+                String.format("%n(==== %s ====)%n", bunName) +
+                String.format("%nPrice: 800,000000%n", expectedPrice);
+        assertEquals(expectedReceipt, burger.getReceipt());
+    }
```
Здесь получилось как-то очень много моков, хотя в принципе для всех тестов достаточно одного мока для булочки, двух для ингредиентов и одного объекта бургера. 
можно улучшить

pomegranate282922 commented on src/test/java/unit_test/IngredientTest.java:
```diff
@@ -0,0 +1,53 @@
+    }
+
+    @Test
+    public void isFillingIngredientType() {
```
Здесь просится параметризация. Можно улучшить

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы.
Нет помника- он нужен обязательно ( напомню, что нужно использовать Java 11 ). 
есть ещё немного замечаний по тестам.

Commit:
После замечаний.
Не понял замечания по поводу
 @Mock
    Burger burger;
Он много где используется.

Commit:
После замечаний.
Не понял замечания по поводу
 @Mock
    Burger burger;
Он много где используется.

Добавил pomНИК (Пришлось спрашивать что это такое))

pomegranate282922 reviewed (commented):
Многое поправил, но осталась пара важных замечаний.

Commit:
В BurgerTest убрал @Mock Burger (Если проблема не в этом, прошу объяснить, что от меня требуется.)

pomegranate282922 reviewed (commented):
Один комментарий пишу уже третий раз и он остаётся неисправленным. Его необходимо поправить.
